pkg/resources/materialized_view_acceptance_test.go

```diff
@@ -0,0 +1,109 @@
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "schema", schemaName),
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "comment", "Terraform test resource"),
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "cluster_by", "[\"ID\",\"DATA\"]"),
+					checkBool("snowflake_materialized_view.test", "is_secure", true),
+				),
+			},
+		},
+	})
+}
+
+func TestAcc_MaterializedView2(t *testing.T) {
+	dbName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	schemaName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	tableName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	viewName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+
+	resource.ParallelTest(t, resource.TestCase{
+		Providers: providers(),
+		Steps: []resource.TestStep{
+			{
+				Config: materializedViewConfig(dbName, schemaName, tableName, viewName, []string{"ID", "DATA"}, fmt.Sprintf("SELECT ID, DATA FROM \\\"%s\\\" WHERE ID LIKE 'foo%%';", tableName)),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "name", viewName),
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "database", dbName),
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "schema", schemaName),
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "comment", "Terraform test resource"),
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "cluster_by", "[\"ID\",\"DATA\"]"),
+					checkBool("snowflake_materialized_view.test", "is_secure", true),
+				),
+			},
+		},
+	})
+}
+
+func materializedViewConfig(dbName string, schemaName string, tableName string, viewName string, c []string, q string) string {
+	// convert the cluster from string slice to string
+	clusters := strings.Join(strings.Split(fmt.Sprintf("%q\n", c), " "), ",")
+	return fmt.Sprintf(`
+resource "snowflake_warehouse" "w" {
+	name = "connection-warehouse"
+}
+
+resource "snowflake_database" "test" {
+	name = "%s"
+}
+
+resource "snowflake_schema" "test" {
+	database  = snowflake_database.test.name
+	name      = "%s"
+}
+
+resource "snowflake_table" "test" {
+	database = snowflake_database.test.name
+	schema   = snowflake_schema.test.name
+	name     = "%s"
+	
+	column {
+		name = "ID"
+		type = "NUMBER(38,0)"
+	}
+		
+	column {
+		name = "DATA"
+		type = "VARCHAR(16777216)"
+	}
+}
+
+resource "snowflake_materialized_view" "test" {
+	name      = "%s"
+	comment   = "Terraform test resource"
+	database  = snowflake_database.test.name
+	schema    = snowflake_schema.test.name
+	is_secure = true
+	or_replace = false
+	cluster_by = %s
+	statement = "%s"
+
+	depends_on = [
+		snowflake_warehouse.w,
+  		snowflake_table.test,
+  	]
+}
+`, dbName, schemaName, tableName, viewName, clusters, q)
```

Comment:
This is off now, looks like you just need name, cluster_by and statement

Reply:
No table and view must be different name.

Reply:
oops sorry misread!